### PR TITLE
Evaluate math expressions in the editor controls

### LINKED ISSUE #234: Allow configurable scaling so Retina displays don't generate enormous files
I tend to create simple gifs for work to illustrate a bug or UX issue I run into on the software we're developing.  Although I use a retina-display Macbook Pro, the resulting gif doesn't need to be ultra crisp and high-resolution.  It merely needs to get the point across.

I tend to use kap to create a gif, and then immediately open that gif up in Photoshop to remove frames from the beginning and end, then scale it to 50% to save on space.  This will often turn a 5mb gif into a ~1mb gif.  It would be great if I didn't need to use photoshop for this process (large gifs tend to create gigantic – multi-gigabite! – scratch disc files on my hard drive as well).

### PR DESCRIPTION
**Pull request by karaggeorge**
Closes #234

@skllcrn will love this one

Kapture 2019-11-28 at 12 53 48

### CODE REVIEW

**Comment by skllcrn** [comment]
You're something else.

File: renderer/containers/editor.js [hunk 1]
```diff
@@ -37,8 +38,21 @@ export default class EditorContainer extends Container {
...
+      this.setState({[name]: value, lastValid: {width, height, ...lastValid}});
+      return;
+    }
+
+    let parsedValue;
+    try {
+      parsedValue = stringMath(value);
+    } catch {
+      parsedValue = null;
+    }
```
Comment (sindresorhus) [inline renderer/containers/editor.js]: Let's prefer `undefined` over `null`.

```js
} catch {}
```

**Comment by sindresorhus** [comment]
I think this should be documented somewhere. Maybe in a "Tip" section in the readme for now? At least until we have https://github.com/wulkano/kap/issues/624.

**Comment by sindresorhus** [comment]
Not related to this change, but I think I found a bug in the inputs, if I write 200, tab to the next and then tab back, it changes.

**Comment by karaggeorge** [comment]
> Not related to this change, but I think I found a bug in the inputs, if I write 200, tab to the next and then tab back, it changes.

I've noticed that too, but I don't know what the expected behavior is. It rounds the dimensions when it does the Math, and rounds the other way again. Maybe if I replaced `Math.round` with `Math.floor`? Not sure if that would guarantee it doesn't change backwards

**Comment by sindresorhus** [comment]
https://github.com/wulkano/kap/pull/769#issuecomment-559647107 still needs to be done, but can be done later.

**Pull request merged**